crates/gateway-types/src/reply.rs

```diff
@@ -642,17 +642,54 @@ pub mod transaction {
     }
 }
 
-/// Used to deserialize replies to StarkNet state update requests.
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum MaybePendingStateUpdate {
```

Comment:
It wasn't quite clear to me what's discriminating the two variants here (`StateUpdate` _always_ has a `block_hash`) -- maybe it would be worth adding a comment about this?